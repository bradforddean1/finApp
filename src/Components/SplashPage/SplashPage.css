.SplashPage {
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	height: 100%;
	padding-left: 20px;
	padding-right: 20px;
}

.SplashPage .cta,
.SplashPage p {
	margin-bottom: 24px;
}

.SplashPage p {
	margin-top: 40px;
}

.SplashPage .demo {
	text-align: center;
}

.SplashPage .credentials {
	display: block;
	margin-top: 8px;
}
