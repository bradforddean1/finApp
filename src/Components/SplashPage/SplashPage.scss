@use "../../scss/variables" as v;

.SplashPage {
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	height: 100%;
	padding-left: 20px;
	padding-right: 20px;
}

@media screen and (min-width: v.$first-break-point) {
	.SplashPage {
		max-width: v.$panel-content-max-width;
		margin: auto;
	}
}

.SplashPage .cta,
.SplashPage p {
	margin-bottom: 24px;
}

.SplashPage p {
	margin-top: 40px;
}

.SplashPage .demo {
	text-align: center;
}

.SplashPage .credentials {
	display: block;
	margin-top: 8px;
}
