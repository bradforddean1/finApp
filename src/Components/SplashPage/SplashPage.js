import React from "react";
import ButtonCta from "../Common/ButtonCta/ButtonCta";
import FullHeaderPage from "../FullHeaderPage/FullHeaderPage";
import "./SplashPage.css";

/**
 * Splash Page Root
 * @component
 */
function SplashPage(props) {
	return (
		<FullHeaderPage>
			<div className="panel">
				<p>
					I am a splash page and I am supposed to tele you about this app...
				</p>

				<ButtonCta className="cta" tag="a" href="/login">
					Go Forth
				</ButtonCta>
			</div>
		</FullHeaderPage>
	);
}

export default SplashPage;
