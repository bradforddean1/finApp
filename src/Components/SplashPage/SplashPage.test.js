import React from "react";
import ReactDOM from "react-dom";
import SplashPage from "./SplashPage";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";

describe("SplashPage Component", () => {
	it("Renders", () => {
		const div = document.createElement("div");
		ReactDOM.render(
			<BrowserRouter>
				<SplashPage />
			</BrowserRouter>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("Renders the Default State", () => {
		const wrapper = shallow(<SplashPage />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
