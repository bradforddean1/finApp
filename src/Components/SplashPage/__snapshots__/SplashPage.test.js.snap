// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashPage Component Renders the Default State 1`] = `
<FullHeaderPage>
  <div
    className="SplashPage"
  >
    <div>
      <p>
        Ztock is here to pull realtime quotes with key financial metrics for domestic stocks. Search for a stock with a valid ticker symbol and you'll get a real time quote with the copanies profile and key financial indicators. You can choose to add the stock to your portfolio to monitor an unlimited number of stocks.
      </p>
      <p>
        You'll need an account to access Ztocks featutres. You can create your own or use the demo account to take a look around.
      </p>
      <div
        className="demo"
      >
        <span
          className="credentials"
        >
          User: demo
        </span>
        <span
          className="credentials"
        >
          Password: testing123
        </span>
      </div>
    </div>
    <ButtonCta
      className="cta"
      href="/login"
      tag="a"
    >
      Go Forth
    </ButtonCta>
  </div>
</FullHeaderPage>
`;
